src/jsdoc.ts

```diff
@@ -213,9 +220,20 @@ export function parseContents(commentText: string): ParsedJSDocComment|null {
         tagName = 'return';
       }
       let type: string|undefined;
-      if (JSDOC_TAGS_BLACKLIST.has(tagName)) {
-        warnings.push(`@${tagName} annotations are redundant with TypeScript equivalents`);
-        continue;  // Drop the tag so Closure won't process it.
+      if (JSDOC_TAGS_INPUT_BLACKLIST.has(tagName)) {
+        if (tagName !== 'template') {
+          // Tell the user to not write blacklisted tags, because there is TS
+          // syntax available for them.
+          warnings.push(`@${tagName} annotations are redundant with TypeScript equivalents`);
+          continue;  // Drop the tag so Closure won't process it.
+        } else {
+          // But @template in particular is special: it's ok for the user to
+          // write it for documentation purposes, but we don't want the
+          // user-written one making it into the output because Closure interprets
+          // it as well.
```

Comment:
Document that for simplicity we don't check that the template type in the "doc-only" comment, name really matches the TS declared template.

Reply:
Done.